lib/Cake/Utility/Hash.php

```diff
@@ -869,10 +872,30 @@ public static function sort(array $data, $path, $dir = 'asc', $type = 'regular')
 		$values = static::extract($result, '{n}.value');
 
 		$dir = strtolower($dir);
-		$type = strtolower($type);
+		$ignoreCase = false;
+		
+		// $type can be overloaded for case insensitive sort
+		if (is_array($type)) {
+
+			if (!empty($type['ignoreCase'])) {
+				$ignoreCase = $type['ignoreCase'];
+			}
+
+			if (!empty($type['ignoreCase'])) {
+				$type = $type['type'];
+			} else {
+				throw new InvalidArgumentException(__d('cake_dev',
+					'Invalid parameter $type. It requires type key to be specified.'
+				));
+			}
+		} else {
+			$type = strtolower($type);
+		}
```

Comment:
This feels like a lot of branching code that could probably be avoided by merging defaults.